src/dispatching/handler.rs

```diff
@@ -1,44 +1,33 @@
 use std::{future::Future, pin::Pin};
 
-use futures::FutureExt;
-
-pub type HandlerResult<E> = Result<(), E>;
-
-/// Asynchronous handler for event `T` (like `&self, I -> Future` fn)
-pub trait Handler<'a, T, E> {
-    fn handle(
-        &self,
+/// An asynchronous handler of a value.
+pub trait Handler<T, E> {
+    #[must_use]
+    fn handle<'a>(
+        &'a self,
         value: T,
-    ) -> Pin<Box<dyn Future<Output = HandlerResult<E>> + 'a>>;
-}
-
-pub trait IntoHandlerResult<E> {
-    fn into_hr(self) -> HandlerResult<E>;
-}
-
-impl<E> IntoHandlerResult<E> for () {
-    fn into_hr(self) -> HandlerResult<E> {
-        Ok(())
-    }
-}
-
-impl<E> IntoHandlerResult<E> for HandlerResult<E> {
-    fn into_hr(self) -> HandlerResult<E> {
-        self
-    }
+    ) -> Pin<Box<dyn Future<Output = Result<(), E>> + 'a>>
+    where
+        T: 'a;
 }
 
-impl<'a, F, Fut, R, T, E> Handler<'a, T, E> for F
+/// The implementation of `Handler` for `Fn(U) -> Future<Output = Result<(),
+/// E>>`.
+///
+/// Looks quite strange for now, but with stabilised asynchronous traits it
+/// should be prettier.
```

Comment:
"stabilised". There aren't any unstable asynchronous traits, so they can't be stabilized...

Reply:
I've removed this.